cloud-sql-java-functions-benchmark/src/main/java/functions/Main.java

```diff
@@ -0,0 +1,185 @@
+import com.zaxxer.hikari.HikariConfig;
+import com.zaxxer.hikari.HikariDataSource;
+import io.opencensus.common.Scope;
+import io.opencensus.exporter.trace.stackdriver.StackdriverTraceConfiguration;
+import io.opencensus.exporter.trace.stackdriver.StackdriverTraceExporter;
+import io.opencensus.trace.Span;
+import io.opencensus.trace.Status;
+import io.opencensus.trace.Tracer;
+import io.opencensus.trace.Tracing;
+import io.opencensus.trace.config.TraceConfig;
+import io.opencensus.trace.samplers.Samplers;
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.util.Properties;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.sql.DataSource;
+import org.apache.commons.lang3.exception.ExceptionUtils;
+
+
+public class Main implements HttpFunction {
+
+  // TODO: replace these values before deploying
+  private static final String PROJECT_ID = "";
+  private static final String CONNECTION_NAME = "";
+  private static final String DB_NAME = "";
+  private static final String DB_USER = "";
+  private static final String DB_PASSWORD = "";
+
+  private static final Logger LOGGER = Logger.getLogger(Main.class.getName());
+  private static final Tracer TRACER = Tracing.getTracer();
+
+  static {
+    try {
+      // Set up StackDriver
+      StackdriverTraceExporter.createAndRegister(
+          StackdriverTraceConfiguration.builder()
+              .setProjectId(PROJECT_ID)
+              .build());
+
+    } catch (java.io.IOException e) {
+      LOGGER.log(Level.WARNING, "Warning: Could not set up Stackdriver Trace", e);
+    }
+
+  }
+
+  @Override
+  public void service(HttpRequest request, HttpResponse response)
+      throws IOException {
+
+    Integer duration = Integer.parseInt(request.getFirstQueryParameter("duration").orElse("5000"));
+    Integer interval = Integer.parseInt(request.getFirstQueryParameter("interval").orElse("500"));
+
+    // Trace every request
+    TraceConfig traceConfig = Tracing.getTraceConfig();
+    traceConfig.updateActiveTraceParams(
+        traceConfig.getActiveTraceParams().toBuilder().setSampler(Samplers.alwaysSample()).build());
+
+    // Set up URL parameters
+    String jdbcURL = String.format("jdbc:postgresql:///%s", DB_NAME);
+    Properties connProps = new Properties();
+    connProps.setProperty("user", DB_USER);
+    connProps.setProperty("password", DB_PASSWORD);
+    connProps.setProperty("socketFactory", "com.google.cloud.sql.postgres.SocketFactory");
+    connProps.setProperty("cloudSqlInstance", CONNECTION_NAME);
+
+    LOGGER.log(Level.INFO, "Started logging.");
+
+    LOGGER.log(Level.INFO, "Started logging.");
+    // Create thread pool
+    ScheduledExecutorService execService = Executors.newScheduledThreadPool(2);
```

Comment:
We probably only want one thread pool per function, so this should probably be static as well.

Reply:
I tried making this static, but then the requests would error out when I try to hit the function multiple times. I think it's because `execService.shutdown()` is called at the end of the request.

Reply:
So don't call `execService.shutdown()` :) 

Some other ways to handle this:
1. I think there's a default executor, which you can use to schedule futures (some what like the default loop for python's asyncio)
2. If not you can init one on start up and then schedule it to shutdown ([Guava has an executor ](https://guava.dev/releases/20.0-rc1/api/docs/com/google/common/util/concurrent/MoreExecutors.html#getExitingExecutorService-java.util.concurrent.ThreadPoolExecutor-)that does this automatically, I think the JDBC socketfactory uses it)

Reply:
Ok, I think I realized why you want to call `execService.shutdown()` when I was rereading the code. I think the. problem here is we actually want to use the `ScheduledFuture` returned from  `execService.scheduleAtFixedRate` and call `cancel` on it. 

Alternatively, we could just schedule some X number of these to happen every Y interval. Up to you (although X might be a reasonable parameter to configure as well).

Reply:
In the last commit I updated this to cancel the `ScheduledFuture` instead of shutting down the thread pool executor. Just to clarify, you're suggesting we add an extra parameter to schedule some number of connection attempts to happen each interval? (instead of the 1/interval we have now)

Reply:
So right now we are doing 1 connect every Y interval in X time. I'm suggesting it might be more useful to do 1 connect every Y interval for a total of X connects. this has the benefit of being that we don't have to worry about leaking connections if the cancel doesn't get called on the scheduled connects.